src/targets/services/helpers/model.js

```diff
@@ -142,11 +157,70 @@ export class Model {
     return shares
   }
 
+  getCompressedShares(nbShares) {
+    const shares = this.getShares(nbShares)
+    return shares.map(share => Model.shareToCompressedBinary(share))
+  }
+
   getBackup() {
     const { occurences, contributions } = this
     return {
       occurences,
       contributions
     }
   }
+
+  getCompressedBackup() {
+    const { occurences, contributions } = this
+    return Model.shareToCompressedBinary({
+      occurences,
+      contributions
+    })
+  }
+
+  static shareToCompressedBinary(share) {
+    const ROWS = vocabulary.length
+    const COLS = Object.keys(classes).length
+    const numberSize = 4
+    const buf = Buffer.alloc((ROWS * COLS + 1) * numberSize)
+
+    buf.writeInt32BE(share.contributions, 0)
+    for (let j = 0; j < ROWS; j++) {
+      for (let i = 0; i < COLS; i++) {
+        buf.writeInt32BE(
+          share.occurences[j][i],
+          (j * COLS + i + 1) * numberSize
+        )
+      }
+    }
+
+    return LZString.compressToBase64(buf.toString('base64'))
```

Comment:
Wasn't it possible to just do `LZString.compressToBase64(JSON.stringify(share))`?

Reply:
It's more compact to write values as 32bits integer and then compress rather than writing them as ascii numerals (could take ~7 * 8bits per value) then compressing

Reply:
I was wondering if `LZString` could handle this by itself, but it seems rather simple, so ok!

Reply:
Btw, there are libs allowing to manipulate matrix, but maybe this is overkill here? https://mathjs.org/docs/datatypes/matrices.html